smartsim/entity/ensemble.py

```diff
@@ -117,19 +112,17 @@ def _initialize_entities(self, **kwargs: t.Any) -> None:
 
         :raises UserStrategyError: if user generation strategy fails
         """
-        strategy = self._set_strategy(kwargs.pop("perm_strat"))
+        strategy = resolve_strategy(kwargs.pop("perm_strat"))
```

Comment:
We are retrieving the perm_strat from kwargs passed in but we have a perm_strat pass to super.__init__, can we retrieve it via the attribute? This is probably more of a miss from my understanding

Reply:
Not a miss of your understanding, it is just very confusingly implemented!

This is a relic from the days before SmartSim was not rigidly using `abc.ABC`s so ensembles inherit from this "psuedo-abstract" `EntityList` class and the idea is that users would implement this `_initialize_entities` function to finish initialization of "abstract" properties.

More than willing move this to named kwarg in this method! Hopefully we will end up trashing this implementation entierly, but if for some reason we do decide to keep it, the "correct" thing to do would be to make `EntityList` a true `abc.ABC`, mark our abstract attrs, and remove the constructor from the interface.

Reply:
Resolving as we have decided to remove the old impl of `Ensemble`